File: ex4_media_ponderada_situacao/main.py
n1 = float(input("Nota 1: "))
p1 = float(input("Peso 1: "))
n2 = float(input("Nota 2: "))
p2 = float(input("Peso 2: "))
n3 = float(input("Nota 3: "))
p3 = float(input("Peso 3: "))

# Média ponderada com parênteses para garantir a ordem correta
media = (n1*p1 + n2*p2 + n3*p3) / (p1 + p2 + p3)

# Uso de operadores relacionais e lógicos (and)
if media >= 7:
    situacao = "Aprovado"
elif media >= 5 and media < 7:   # expressão lógica com 'and'
    situacao = "Recuperação"
else:
    situacao = "Reprovado"

print(f"Média ponderada: {media:.2f}")
print(f"Situação: {situacao}")

File: ex4_media_ponderada_situacao/main.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    double n1, p1, n2, p2, n3, p3;

    cout << "Nota 1: "; cin >> n1;
    cout << "Peso 1: "; cin >> p1;
    cout << "Nota 2: "; cin >> n2;
    cout << "Peso 2: "; cin >> p2;
    cout << "Nota 3: "; cin >> n3;
    cout << "Peso 3: "; cin >> p3;

    double media = (n1*p1 + n2*p2 + n3*p3) / (p1 + p2 + p3);

    string situacao;
    if (media >= 7.0) {
        situacao = "Aprovado";
    } else if (media >= 5.0 && media < 7.0) { // && para AND
        situacao = "Recuperação";
    } else {
        situacao = "Reprovado";
    }

    cout << fixed << setprecision(2);
    cout << "Média ponderada: " << media << endl;
    cout << "Situação: " << situacao << endl;

    return 0;
}